docs/oci-storage.md

```diff
@@ -1,40 +1,152 @@
+* **GitRepo-level secret (ociRegistrySecret):** – Overrides the global secret for a specific GitRepo.  
+  * This is a user-defined secret that you reference in the `GitRepo` resource. 
+
+:::note
+If the referenced secret doesn't exist or contains invalid credentials, Fleet logs an error and skips deployment. It doesn't fall back to etcd.
+:::
+
+### GitRepo Configuration Example
+
+The GitRepo spec is extended with information about the OCI Registry. You must update your GitRepo resource to include a reference to the secret:
+
+```yaml
+metadata:
+  name: coffee-shop-deployments
+  namespace: fleet-local
+spec:
+  repo: https://github.com/example/coffee-shop-fleet
+  branch: main
+  paths:
+    - ./manifests
+  ociRegistrySecret: ocistorage
+```
+
+The `ociRegistrySecret` references a secret in the same namespace, which contains the address and credentials needed to read/write to the OCI registry. 
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  reference: <base64-encoded-registry-url>
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+# Optional
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
+```
+:::note
+If your user is not defined in the spec of the Repo.Then Fleet tries to use default credentials.
+:::
+
+You should follow these best practices:
+
+* Use read-only `agentUsername` and `agentPassword` credentials for agents to enhance security.  
+  * However, if you don’t send these credentials, the agent uses user credentials with read/write permissions.  
+* Insecure TLS defaults to `false`, and http is disabled by default.   
+  * Fleet allows these flags for development and testing purposes, but they should never be used in production.  
+  * If you use these tags, you expose your app to security vulnerabilities, and put cluster workloads, and credentials at a risk of tampering with.  
+* Changing the secret does not update the deployments, only on the next GIT update it would use the new storage registry.
+
+
+Fleet uses `gitcredential` secret when the `clientSecretName` field is empty in a GitRepo. It uses global `ocistorage` as a fallback if `ociRegistrySecret` is not defined.
+
+:::note
+Fleet checks for the integrity of OCI artifacts and Fleet tags OCI artifact as Latest
+:::
+
+### Fleet Example
+
+Consider the GitRepo file for fleet-examples repository. 
+
+```yaml
+apiVersion: fleet.cattle.io/v1alpha1
+kind: GitRepo
+metadata:
+ name: frontend-oci
+ namespace: fleet-local
+spec:
+ repo: https://github.com/your-org/fleet-oci-example.git
+ branch: main
+ paths:
+ - ./frontend/frontend-deployment.yaml
+    - ./frontend/frontend-service.yaml
+  ociRegistrySecret: ocistorage
+```
+
+To define OCI Secret, you must create an oci-secret.yaml file
```

Comment:
This is not completely true.
You also mention that you can also create the secret using `kubectl`.
So I would no say that you `must` create the yaml... you have the option to create the yaml or not.